### ISSUE TITLE
Backspace issue in peek definition window

### ISSUE THREAD

**Issue by pizza2code** [VsVim/VsVim#2492]
In a peek definition window, backspace does not immediately update the displayed command buffer in normal mode.

Example:

1. Open peek definition window
2. Typo `:wqa`
3. Backspace once or twice
4. The command buffer still displays `:wqa`

The displayed command only changes after focus leaves the peek definition window, or after the entire command is removed.

Using Visual Studio 2015 and the latest VsVim (2.6.0).

**Comment by ricksladkey** [comment]
I can reproduce this bug with VS2019 and the head of master.

**Comment by ricksladkey** [comment]
I did some debugging and the problem here is that the command margin associated with a peek window does not receive the backspace key through the normal WPF keyboard dispatching method. Instead VsVim receives it via an OLE command target, which routes it through the corresponding vim buffer, which then forwards it to command mode, which then forwards it to the history session, where it does in fact the history command being built, but never ends up reflected in the context of the command margin's text box. Phew.

In short, the command margin's text box input is getting running afoul of Visual Studio's legacy command processing, but only when that command margin is associated with a peek window. It's not immediately clear to me how to fix this, but the scenario was complex enough that I wanted to report the results of my investigation for posterity.

I will look at this some more when I get a chance.

**Comment by ricksladkey** [comment]
This seems to me to be Visual Studio bug, and my work around is not pretty, but it is very specific and only fixes a problem that should not be occurring. The set of keys that misbehave can easily be added to if more cases are found.
